rollup/internal/controller/sender/sender.go

```diff
@@ -373,13 +397,63 @@ func (s *Sender) resubmitTransaction(tx *gethTypes.Transaction, baseFee uint64)
 		txInfo["adjusted_gas_tip_cap"] = gasTipCap.Uint64()
 		txInfo["original_gas_fee_cap"] = originalGasFeeCap.Uint64()
 		txInfo["adjusted_gas_fee_cap"] = gasFeeCap.Uint64()
+
+	case BlobTxType:
+		originalGasTipCap := tx.GasTipCap()
+		originalGasFeeCap := tx.GasFeeCap()
+		originalBlobGasFeeCap := tx.BlobGasFeeCap()
+
+		// bumping at least 100%
+		gasTipCap := new(big.Int).Mul(originalGasTipCap, big.NewInt(2))
+		gasFeeCap := new(big.Int).Mul(originalGasFeeCap, big.NewInt(2))
+		blobGasFeeCap := new(big.Int).Mul(originalBlobGasFeeCap, big.NewInt(2))
+
+		// adjust for rising basefee
+		currentGasFeeCap := getGasFeeCap(new(big.Int).SetUint64(baseFee), gasTipCap)
+		if gasFeeCap.Cmp(currentGasFeeCap) < 0 {
+			gasFeeCap = currentGasFeeCap
+		}
+
+		// but don't exceed maxGasPrice
+		if gasFeeCap.Cmp(maxGasPrice) > 0 {
+			gasFeeCap = maxGasPrice
+		}
+
+		// gasTipCap <= gasFeeCap
+		if gasTipCap.Cmp(gasFeeCap) > 0 {
+			gasTipCap = gasFeeCap
+		}
+
+		// adjust for rising blobbasefee
+		currentBlobGasFeeCap := getBlobGasFeeCap(new(big.Int).SetUint64(blobBaseFee))
+		if blobGasFeeCap.Cmp(currentBlobGasFeeCap) < 0 {
+			blobGasFeeCap = currentBlobGasFeeCap
+		}
+
+		// but don't exceed maxBlobGasPrice
+		if blobGasFeeCap.Cmp(maxBlobGasPrice) > 0 {
+			blobGasFeeCap = maxBlobGasPrice
+		}
+
+		feeData.gasFeeCap = gasFeeCap
+		feeData.gasTipCap = gasTipCap
+		feeData.blobGasFeeCap = blobGasFeeCap
+		txInfo["original_gas_tip_cap"] = originalGasTipCap.Uint64()
+		txInfo["adjusted_gas_tip_cap"] = gasTipCap.Uint64()
+		txInfo["original_gas_fee_cap"] = originalGasFeeCap.Uint64()
+		txInfo["adjusted_gas_fee_cap"] = gasFeeCap.Uint64()
+		txInfo["original_blob_gas_fee_cap"] = originalBlobGasFeeCap.Uint64()
+		txInfo["adjusted_blob_gas_fee_cap"] = blobGasFeeCap.Uint64()
+
+	default:
+		return nil, fmt.Errorf("unsupported transaction type: %s", s.config.TxType)
 	}
 
 	log.Info("Transaction gas adjustment details", "service", s.service, "name", s.name, "txInfo", txInfo)
 
 	nonce := tx.Nonce()
 	s.metrics.resubmitTransactionTotal.WithLabelValues(s.service, s.name).Inc()
-	tx, err := s.createAndSendTx(&feeData, tx.To(), tx.Value(), tx.Data(), &nonce)
+	tx, err := s.createAndSendTx(&feeData, tx.To(), tx.Data(), tx.BlobTxSidecar(), &nonce)
```

Comment:
Will this work in the graceful restart path (inside `checkPendingTransaction`)? Or do we need to add the blob data to `PendingTransaction`?

Reply:
Yes. Because the db only uses rlp encoding/decoding to handle transactions (the `rlp_encoding` field), other fields related to transaction info are just added for debugging.

btw, resubmission is tested in unit tests now because each unit test needs to wait until the transactions are confirmed, and many tests would trigger this logic (both explicitly added and implicitly resubmitted).